examples/data_augmentation/reinforcement/main.py

```diff
@@ -0,0 +1,350 @@
+    help="Name of the pre-trained augmentation model checkpoint to load.")
+parser.add_argument(
+    '--num-aug', type=int, default=4,
+    help="number of augmentation samples when fine-tuning aug model")
+parser.add_argument(
+    '--classifier-pretrain-epoch', type=int, default=10,
+    help="number of epochs to pretrain the classifier")
+args = parser.parse_args()
+
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+logging.root.setLevel(logging.INFO)
+
+
+class RLAugmentClassifierTrainer:
+    def __init__(self):
+        self._prepare_data_iterator()
+        self._init_aug_model()
+        self._init_classifier()
+
+    def _prepare_data_iterator(self):
+        tx.utils.maybe_create_dir(args.output_dir)
+
+        # Loads data
+        num_train_data = config_data.num_train_data
+        self.num_train_steps = \
+            int(num_train_data / config_data.train_batch_size *
+                config_data.max_train_epoch)
+
+        train_dataset = tx.data.RecordData(
+            hparams=config_data.train_hparam, device=device)
+        val_dataset = tx.data.RecordData(
+            hparams=config_data.eval_hparam, device=device)
+        test_dataset = tx.data.RecordData(
+            hparams=config_data.test_hparam, device=device)
+        self.iterator = tx.data.DataIterator(
+            {"train": train_dataset,
+             "eval": val_dataset,
+             "test": test_dataset}
+        )
+
+        self.val_data_iterator = tx.data.DataIterator({"eval": val_dataset})
+        self.val_data_iterator.switch_to_dataset("eval")
+
+    def _init_aug_model(self):
+        # pylint: disable=protected-access
+        # Builds data augmentation BERT
+        aug_model = BertForMaskedLM.from_pretrained(
+            args.augmentation_model_name)
+        aug_model.to(device)
+        aug_tokenizer = tx.data.BERTTokenizer(
+            pretrained_model_name=args.augmentation_model_name)
+        input_mask_ids = aug_tokenizer._map_token_to_id('[MASK]')
+        # Builds augmentation optimizer
+        aug_lr = 4e-5
+        param_optimizer = list(aug_model.named_parameters())
+        no_decay = ['bias', 'LayerNorm.bias', 'LayerNorm.weight']
+        optimizer_grouped_parameters = [
+            {'params': [p for n, p in param_optimizer if
+                        not any(nd in n for nd in no_decay)],
+             'weight_decay': 0.01},
+            {'params': [p for n, p in param_optimizer if
+                        any(nd in n for nd in no_decay)],
+             'weight_decay': 0.0}]
+        aug_optim = tx.core.BertAdam(
+            optimizer_grouped_parameters, betas=(0.9, 0.999),
+            eps=1e-6, lr=aug_lr)
+        # Builds data augmentation wrapper
+        self.aug_wrapper = MetaAugmentationWrapper(
+            aug_model, aug_optim, input_mask_ids, device, args.num_aug)
+
+    def _init_classifier(self):
+        # Builds downstream BERT classifier
```

Comment:
Downstream BERT would sound a bit confusing. Probably call it BERT for classification task?